app/helpers/application_helper.rb

```diff
@@ -9,13 +9,20 @@ def configured_organization_name
   def parse_application_data(data)
     questions = []
     data.each do |k, v|
-      match_data = k.match(/^(prompt|response|data_type)_(\d+)$/)
+      match_data = k.match(/^(prompt|response|data_type)?_?(\d+)$/)
       next unless match_data.present?
-      input_type, number = match_data.captures
-      questions[number.to_i] ||= []
-      input_types = %w(prompt response data_type)
-      index = input_types.index input_type
-      questions[number.to_i][index] = v
+      captures = match_data.captures.select{|c| !c.nil? }
+      if captures.length == 2
+        input_type, number = match_data.captures
+        questions[number.to_i] ||= []
+        input_types = %w(prompt response data_type)
+        index = input_types.index input_type
+        questions[number.to_i][index] = v
+      elsif captures.length == 1
+        q_id, number = v.split('-')
+        questions[number.to_i] ||= []
+        questions[number.to_i][3] = q_id.to_i + 1
```

Comment:
I don't see the issue. If they're deleted, then it just won't read from the hash? If the question is deleted, then the data associated with said question is irrelevant as well. Simply associating based on question name is a poor idea because there can be different contexts outside of the question. For example:

**As a child,**

 * Life goal?
 * Favorite super hero?

**As you are now**

 * Life goal?
 * Favorite super hero?

There's no way to differentiate between the two besides their ID. They will have the same question prompt and data type. You can't go based off the 'context header' part because in our system those headers are **also** questions.

I don't see a solution that can work for every question other than this.